datadog/dogstatsd/base.py

```diff
@@ -274,24 +297,66 @@ def __init__(
         self._reset_telemetry()
         self._telemetry_flush_interval = telemetry_min_flush_interval
         self._telemetry = not disable_telemetry
+        self._last_flush_time = time.time()
 
+        self._current_buffer_total_size = 0
+        self._buffer = []  # type: List[Text]
         self._buffer_lock = RLock()
 
+        self._reset_buffer()
+
+        # This lock is used for backwards compatibility to prevent concurrent
+        # changes to the buffer when the user is managing the buffer themselves
+        # via deprecated `open_buffer()` and `close_buffer()` functions.
+        self._manual_buffer_lock = RLock()
+
+        # If buffering is disabled, we bypass the buffer function.
+        self._send = self._send_to_buffer
+        if disable_buffering is True:
```

Comment:
```suggestion
        if disable_buffering:
```

Reply:
Fixed in https://github.com/DataDog/datadogpy/pull/670/commits/1c83052e6b4990d0240c180b17e416937cfa8cad